server/api.go

```diff
@@ -402,6 +394,37 @@ func (a *API) getConnectedChannels(w http.ResponseWriter, r *http.Request) {
 	a.writeJSON(w, http.StatusOK, paginatedLinks)
 }
 
+func (a *API) getMSTeamsTeamList(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodOptions {
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	teams, err := a.p.GetMSTeamsTeamList(userID)
+	if err != nil {
+		http.Error(w, "Error occurred while fetching the MS Teams team list.", http.StatusInternalServerError)
+		return
+	}
+
+	sort.Slice(teams, func(i, j int) bool {
+		return teams[i].ID < teams[j].ID
+	})
+
+	offset, limit := a.p.GetOffsetAndLimitFromQueryParams(r)
```

Comment:
Here in GetOffsetAndLimitFromQueryParams method we only need the query object from request-response why are we sending the whole request object ?